src/index.js

```diff
@@ -0,0 +1,87 @@
+import './style.css';
+
+const submit = document.querySelector('.add-btn');
+const tasksDiv = document.querySelector('.tasks');
+const taskInput = document.querySelector('.add-task input');
+
+let editId;
+let isEditTask = false;
+let tasks = JSON.parse(localStorage.getItem('task-list'));
+
+window.editTask = (taskId, taskDescripton) => {
+  editId = taskId;
+  isEditTask = true;
+  taskInput.value = taskDescripton;
+};
+
+const addElemToPage = () => {
+  let div = '';
+  if (tasks) {
+    tasks.forEach((task, index) => {
+      div += `
+      <div class='task' data-id='task.id'>
+        <input id="${index}" class='checkbox' onclick='updateCompleted(this)' type="checkbox">
+        <p id='${index}'class='text' contenteditable="false">${task.descripton}</p>
+        <button type="button"  id="${index}" class='edit' >
+        <span onclick='editTask(${index}, "${task.descripton}")' class="material-symbols-outlined">edit</span>
+        </button>
+        <button type="button"  id="${index}" class='del' onclick='del(${index})' >
+        <span class='material-symbols-outlined del'>delete</span>
+        </button>
+        </div>
+      `;
+    });
+  }
+  tasksDiv.innerHTML = div;
+};
+addElemToPage();
+
+submit.addEventListener('click', () => {
+  const userTask = taskInput.value;
+  if (!isEditTask) {
+    if (!tasks) {
+      tasks = [];
+    }
+    const task = {
+      id: Date.now(),
+      descripton: userTask,
+      completed: false,
+    };
+    tasks.push(task);
+  } else {
+    isEditTask = false;
+    tasks[editId].descripton = userTask;
+  }
+  taskInput.value = '';
+  localStorage.setItem('task-list', JSON.stringify(tasks));
+  addElemToPage();
+});
```

Comment:
- [ ] Please know that you should have a property called index in your task object, this index should be set to the value of the new array length each time you add a new task (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5), right now in your local storage you don't have that:
![image](https://user-images.githubusercontent.com/20567503/195326276-afd3c1c3-0ec3-456f-b10a-6442144d1e11.png)

- [ ] Please also take in mind while adding that index property that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)

Reply:
Hi @mirouhml, Thanks a lot 👍 for your efforts in reviewing my code ☺️.